syntax = "proto3";

package fuzzermsgs;

message BuildRequest {
    string package_name = 1;
    string engine = 2;
    string sanitizer = 3;
    string ossfuzz = 4;
    string source_path = 5;
}

message BuildOutput {
    string package_name = 1;
    string engine = 2;
    string sanitizer = 3;
    string output_ossfuzz_path = 4;
    string source_path = 5;
}

message WeightedTarget {
    float weight = 1;
    BuildOutput target = 2; 
    string harness_path = 3;
}

message Crash {
    BuildOutput target = 1;
    string harness_path = 2;
    bytes crash_input = 3;
}